Numerical_solution_of_SEIRD.Rmd

```diff
@@ -0,0 +1,356 @@
+over time—thus, as $\Delta t$ is allowed to get larger, the approximation of
+$g(t, f)$ by a constant value on an interval of width $\Delta t$ becomes more
+prone to error, which ultimately accumulates in the approximate solution and
+causes it to deviate from the true values.
+
+However, obtaining a satisfactory solution to the differential equation is not
+as simple as setting $\Delta t$ to some minuscule value, as decreasing the 
+value of $\Delta t$ will cause the runtime of the method to increase, and for
+very small values of $\Delta t$, the runtime may be prohibitive.
+
+### Effect of solver step size on solutions
+
+Before looking at the epidemiological model, we examine the numerical error
+arising from the solver step size in a simple model for which we have an
+analytical solution. The differential equation is:
+
+$$\frac{df}{dt} = 2t;\;\;\;\; f(t=0)=0$$
+
+We solve this differential equation on the interval from $t=0$ to $t=3$ with
+two different choices of the time step ($\Delta t=1$ and $\Delta t=0.1$). We 
+compare the obtained approximate solutions with the known true solution.
+
+```{r, fig.width=7, fig.height=5}
+rhs <- function(t, state, params){
+  list(2*t)
+}
+
+true_solution <- function(t){
+  t ^ 2
+}
+
+
+sparse_times <- seq(0, 3, by=1)
+dense_times <- seq(0, 3, by=0.1)
+
+
+# Solve using "euler" from deSolve
+# It uses the same times on which output is requested as the solver time points
+numerical_solution_sparse = ode(
+  y=0,
+  times=sparse_times,
+  parms=NULL,
+  func=rhs,
+  method="euler",
+)
+
+numerical_solution_dense = ode(
+  y=0,
+  times=dense_times,
+  parms=NULL,
+  func=rhs,
+  method="euler",
+)
+
+
+# Save the results in data frames for plotting
+results_sparse <- data.frame(
+  t=numerical_solution_sparse[,1],
+  f=numerical_solution_sparse[,2],
+  method="Numerical, Δt=1"
+)
+
+results_dense <- data.frame(
+  t=numerical_solution_dense[,1],
+  f=numerical_solution_dense[,2],
+  method="Numerical, Δt=0.1"
+)
+
+results_true <- data.frame(
+  t=seq(0, 3, by=0.01),
+  f=true_solution(seq(0, 3, by=0.01)),
+  method="True solution"
+)
+
+
+ggplot() +
+  geom_point(data=rbind(results_sparse, results_dense),
+             aes(x=t, y=f, color=method),
+             show.legend=F) +
+  geom_line(data=rbind(results_sparse, results_dense, results_true),
+            aes(x=t, y=f, color=method))
+```
+
+We observe that the large step size of $\Delta t=1$ causes significant
+inaccuracy in the solution. In particular, because the derivative is increasing
+over time, the forward Euler method (which, between solver time steps, assumes
+that the  derivative remains constant at an old value) produces an approximate
+solution that is too low. For the smaller time step $\Delta t=0.1$, the
+approximate solution is more accurate, although a negative deviation is still
+apparent.
+
+## Effect of solver step size on the SEIR model
+
+First, we load the SEIRD model from comomodels, and specify plausible parameter
```

Comment:
SEIR and SEIRD

Reply:
Not going to highlight every one of these, might be easier for you to just do search and replace if appropriate.